src/Console/Commands/JsonApiGeneratorCommand.php

```diff
@@ -0,0 +1,250 @@
+     *
+     * @var string
+     */
+    protected $subNamespace;
+
+    /**
+     * Whether generated files should be grouped by their resource files.
+     *
+     * @var mixed
+     */
+    protected $namespaceByResource;
+
+    /**
+     * The location of all generator stubs
+     *
+     * @var string
+     */
+    private $stubsDirectory = __DIR__.'/../../../stubs';
+
+    /**
+     * Create a new config clear command instance.
+     *
+     * @return void
+     */
+    public function __construct(Filesystem $files)
+    {
+        parent::__construct($files);
+
+        $this->useEloquent = config('json-api.generator.use-eloquent');
+        $this->subNamespace = config('json-api.generator.namespace');
+        $this->namespaceByResource = config('json-api.generator.by-resource');
+    }
+
+    /**
+     * Build the class with the given name.
+     * Remove the base controller import if we are already in base namespace.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    protected function buildClass($name)
+    {
+        $namespace = $this->getNamespace($name);
+
+        $stub = $this->files->get($this->getStub());
+
+        $this->replaceNamespace($stub, $name)
+                ->replaceResourceType($stub, $this->getResourceName());
+
+        return $stub;
+    }
+
+    /**
+     * Replace the value of the resource type constant
+     *
+     * @param mixed $stub
+     * @param mixed $resource
+     */
+    protected function replaceResourceType(&$stub, $resource)
+    {
+        $stub = str_replace(
+            'dummyResourceType', camel_case($resource), $stub
+        );
+
+        return $this;
+    }
+
+    /**
+     * Get the resource name
+     *
+     * @return string
+     */
+    protected function getResourceName()
+    {
+        $name = ucwords($this->argument('resource'));
+
+        if($this->namespaceByResource) {
+            return str_plural($name);
+        }
+
+        return $name;
+    }
+
+    /**
+     * Get the desired class name from the input.
+     *
+     * @return string
+     */
+    protected function getNameInput()
+    {
+        if( ! $this->namespaceByResource) {
+            return $this->getResourceName();
+        }
+
+        return $this->type;
+    }
+
+    /**
+     * Get the stub file for the generator.
+     *
+     * @return string
+     */
+    protected function getStub()
+    {
+        if($this->isIndependent) {
+            return $this->getStubFor('independent');
+        }
+
+        if ($this->isEloquent()) {
+            return $this->getStubFor('eloquent');
+        }
+
+        return $this->getStubFor('abstract');
+    }
+
+    /**
+     * Get the stub for specific generator type
+     *
+     * @param string mentationType
+     */
+    private function getStubFor($implementationType)
+    {
+        return implode('', [
+            $this->stubsDirectory,
+            '/',
+            $implementationType,
+            '/',
+            $this->type,
```

Comment:
This needs to be `lcfirst($this->type)` otherwise it fails to find the file on systems that have case-sensitive file names